### PR TITLE
drivers: stm32_i2c: reuse already computed settings and few minor changes

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/include/drivers/stm32_i2c.h [hunk 1]
```diff
@@ -120,6 +122,8 @@ struct i2c_handle_s {
 	unsigned long clock;
 	enum i2c_state_e i2c_state;
 	uint32_t i2c_err;
+	uint32_t saved_timing;
+	uint32_t saved_frequency;
```
Comment (jforissier) [inline core/include/drivers/stm32_i2c.h:120]: Should this be `unsigned long`? (just wondering since `stm32_clock_get_rate()` returns `unsigned long`)

Comment (etienne-lms) [reply core/include/drivers/stm32_i2c.h:120]: that's right, thanks, will fix.

File: core/include/drivers/stm32_i2c.h [hunk 2]
```diff
@@ -110,7 +110,9 @@ struct i2c_cfg {
  * @clock: clock ID
  * @i2c_state: Driver state ID I2C_STATE_*
  * @i2c_err: Last error code I2C_ERROR_*
- * @sec_cfg: I2C regsiters configuration storage
+ * @saved_timing: Saved timing value if already computed
+ * @saved_frequency: Saved frequency value if already computed
+ * @sec_cfg: I2C registers configuration storage
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/include/drivers/stm32_i2c.h:109]: oh, sorry, yes this fixes a typo. I'll move to another patch in the series.

File: core/drivers/stm32_i2c.c [hunk 3]
```diff
@@ -587,13 +596,16 @@ static int i2c_setup_timing(struct i2c_handle_s *hi2c,
...
 	}
 
-	DMSG("I2C Speed Mode(%i), Freq(%i), Clk Source(%i)",
+	DMSG("I2C Speed Mode(%i), Freq(%"PRIu32"), Clk Source(%lu)",
 	     init->speed_mode, i2c_specs[init->speed_mode].rate, clock_src);
-	DMSG("I2C Rise(%i) and Fall(%i) Time",
+	DMSG("I2C Rise(%"PRId32") and Fall(%"PRId32") Time",
 	     init->rise_time, init->fall_time);
-	DMSG("I2C Analog Filter(%s), DNF(%i)",
+	DMSG("I2C Analog Filter(%s), DNF(%"PRIu8")",
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c]: Right, I'll move to the patch on I2C trace messages.

File: core/drivers/stm32_i2c.c [hunk 4]
```diff
@@ -587,13 +596,16 @@ static int i2c_setup_timing(struct i2c_handle_s *hi2c,
 		return rc;
 	}
 
-	DMSG("I2C Speed Mode(%i), Freq(%i), Clk Source(%i)",
+	DMSG("I2C Speed Mode(%i), Freq(%"PRIu32"), Clk Source(%lu)",
 	     init->speed_mode, i2c_specs[init->speed_mode].rate, clock_src);
-	DMSG("I2C Rise(%i) and Fall(%i) Time",
+	DMSG("I2C Rise(%"PRId32") and Fall(%"PRId32") Time",
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c:636]: Right, I'll move to the patch on I2C trace messages.

File: core/drivers/stm32_i2c.c [hunk 5]
```diff
@@ -587,13 +596,16 @@ static int i2c_setup_timing(struct i2c_handle_s *hi2c,
 		return rc;
 	}
 
-	DMSG("I2C Speed Mode(%i), Freq(%i), Clk Source(%i)",
+	DMSG("I2C Speed Mode(%i), Freq(%"PRIu32"), Clk Source(%lu)",
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c]: Right, I'll move to the patch on I2C trace messages.

File: core/drivers/stm32_i2c.c [hunk 6]
```diff
@@ -561,20 +561,29 @@ static int i2c_setup_timing(struct i2c_handle_s *hi2c,
...
+	}
+
 	do {
 		rc = i2c_compute_timing(init, clock_src, timing);
 		if (rc) {
 			EMSG("Failed to compute I2C timings");
 			if (init->speed_mode > I2C_SPEED_STANDARD) {
 				init->speed_mode--;
-				IMSG("Downgrade I2C speed to %uHz)",
+				IMSG("Downgrade I2C speed to %"PRIu32"Hz)",
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c:621]: Right, I'll move to the patch on I2C trace messages.

File: core/drivers/stm32_i2c.c [hunk 7]
```diff
@@ -561,20 +561,29 @@ static int i2c_setup_timing(struct i2c_handle_s *hi2c,
 			    uint32_t *timing)
 {
 	int rc = 0;
-	uint32_t clock_src = stm32_clock_get_rate(hi2c->clock);
+	unsigned long clock_src = stm32_clock_get_rate(hi2c->clock);
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c]: ok, i'll move to a specific change.

**Comment by etienne-lms** [comment]
Updated with fixup commits inserted.
Maybe easier to review if I squash them?

File: core/include/drivers/stm32_i2c.h (hunk 1 above)
Comment (etienne-lms) [reply core/include/drivers/stm32_i2c.h]: addressed in the fixup change following this change.

File: core/drivers/stm32_i2c.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c:621]: This is related to this commit "drivers: stm32_i2c: allow any bus frequency" as it changes argument from `i2c_specs[init->speed_mode].rate` to `init->bus_rate`.

File: core/drivers/stm32_i2c.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c:638]: done

File: core/drivers/stm32_i2c.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_i2c.c:636]: done

File: core/include/drivers/stm32_i2c.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/include/drivers/stm32_i2c.h:109]: done

File: core/drivers/stm32_i2c.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (jforissier) [reply core/drivers/stm32_i2c.c:621]: Right

**Comment by jforissier** [comment]
LGTM, you can add my Ack while squashing, I will do a quick last chack and shout if something is wrong ;-)
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
Ok, thanks.

**Comment by etienne-lms** [comment]
Fixup commit squashed.
Tag applied.

**Pull request merged**